### DISCUSSION TITLE
Status of Light

### DISCUSSION THREAD

**Discussion by Damageplan1980** [andesse/hue-clip-api.node-red-flows#6] [Q&A] [answered]
Hi there,

first of all: Thanks for your work, very appreciated :)

I'm using you contribution to trigger a handfull of HUE lights over my KNX Bus at home.
Everythings working right now except for one thing:
When I switch lights via the HUE app on my phone, the KNX Bus does not get the status of the light.
Therefore the status LEDs on my KNX switches show the wrong value because they dont know that the light was switched on from the HUE app.

How can I get the status of lights to send it to the KNX Bus?

Thanks for your help
Matthias

**Comment by andesse** [comment, marked as answer]
@Damageplan1980 
Hi, thanks for the feedback. 

For the status of your Lamps / Rooms you need to connect the Orange Event node to a switch node. The switch need to contain the ID of the lamp/room. (You can discover it with the helper function flow, if you don’t have it from the first release, download it separately check in readme) 

Connect the switch node to the „HUE light / grouped_light Event Filter“ Subflow. (When you have a look in the repository, you can see the setup on the top right in the picture)

The orange event node fires all events out, goes into the switch, the lamp get filtered out and the subflow filters true / true & false / false on the outputs. Generally you need the middle output. 

For ever lamp / group you have to set it up separately. 

if you don’t get any events out of the orange event flow node, you can restart node red. 
if you have problems with this node, some people have for unknown reason, download the SSE_Event_Stream Flow. This one does the same. (Check also Readme)

The descriptions are also in the label nodes. DOUBLE CLICK label 

hope this helps. :)

greetings Andreas

**Reply by Damageplan1980** [reply]
@andesse 
Hi Andreas,

wow, thanks for the super fast answer, I'm feeling a bit dump right now :D
"Wer lesen kann, ist klar im Vorteil"... but as an excuse, I'm pretty new to this whole Node-Red thing ;)

And I'm also pretty shure that there will be more questions to come, when I try to implement my few Hue lights with color temperature and RGBW ;)

Thanks
Matthias

**Comment by andesse** [comment]
@Damageplan1980 
ich nehm das hier ernst :) 
Mit den RGB Werten ist es einfach. Die müssen als JSON übergeben werden. Auf der linken Seite des Flows sind die ganzen gelben Nodes. 
da steht direkt drauf was es macht. Dann kopierst Du eine Node. 

Du klickst dran drauf, gehst in den Teil mit dem Json, siehst den Code und kannst dann wahrscheinlich direkt nachvollziehen, wie die Werte bearbeite werden müssen, damit das heraus kommt was Du möchtest. 
Da ist auch wieder ein DOUBLECLICK :)

Gruß

**Reply by Damageplan1980** [reply]
@andesse 
Ich merks ;)
Schon wirklich sehr geil, wie gut das funktioniert. Hatte meine ersten Gehversuche mit Node-Red und HUEMAGIC gemacht und war direkt frustriert, weil ich gleich mal über den Bug gestolpert bin, wo die Lampe dann plötzlich nicht mehr schaltet.
Ich taste mich jetzt einfach mal weiter an das ganze Thema ran...

Was hat es mit der 3.2 auf sich? Warum ne Alternative zu den Event-Streams? Irgendwelche Vorteile dadurch?

Danke und dir und viele Grüße
Matthias

**Reply by andesse** [reply]
@Damageplan1980 
Keine wirklichen Vorteile. Ein paar Leute hatten mal Probleme mit der Contrib Hue-Events. Dies ließ sich dann aber durch anlegen eines neuen Benutzers beheben (api-key, Taste drücken an der Bridge)

Die SSE (Server Side Events) Node contrib ist im Vergleich zur HUE Event contrib eine  universell einsetzbare Node. Nicht nur für HUE. 

Das ist einfach gesagt eine HTTP Node die eine Verbindung zum Endpunkt aufrecht erhalten kann (timeout kann man dort definieren, aber auch, das es sich sofort neu verbindet). Das geht mit der In NodeRed vorhandenen http Node leider nicht. 

Die Hue-Event contrib hat den Vorteil das es für Hue abgestimmt ist und auch die pair Funktion für neue API keys hat. 

Aber bei beiden Varianten kommt hinten der gleiche payload raus. 

Es ist eine Backup Lösung. Ich möchte mit meiner Repository ALLE Mittel zur Verfügung stellen die im Notfall nötig sein könnten um ein funktionierendes System zu haben (auch mit einer schnellen Reaktionszeit bei fragen Leuten helfen). 

Das ist mir wichtig, da das Licht im Smarthome mmm. die wichtigste Komponente ist und laufen muss. Ich kann die Trägheit der Programmierer anderer contribs nicht verstehen. Deshalb mit so wenig contribs wie möglich das Ding selbst in die Hand nehmen, dann läuft es stabil. 

Ich weiß aus eigener Erfahrung wie gestresst und verärgert die Ehefrau sein kann, wenn im Badezimmer (bei uns im Altbau innenliegenden, ohne Fenster) das Licht nicht eingeschaltet werden kann. 
HORROR 🥶🥶🤣🤣

Ich bin im HUE Beta Programm für Firmware der Bridges und habe einen Developer Account bei HUE. Ich werde hier bei Änderungen sofort neue Tools / Flows und Hinweise so schnell es geht zur Verfügung stellen. 

Danke übrigens nochmal fürs Feedback.
Das ist wichtig um weitere Anpassungen zu machen damit es noch einfacher wird. 
Ggf. mache ich demnächst mal eine Version mit deutscher Readme und KLICKMICH in dem Flow ;)

Gruß und viel Spaß beim programmieren 
Andreas
